src/components/molecules/subheaderList/SubheaderList.tsx

```diff
@@ -0,0 +1,48 @@
+import { ListSubheader, ListItemButton, ListItemText } from '@mui/material';
+import { Link } from 'react-router-dom';
+
+interface ListButtonProps {
+  label: string;
+  filter: string;
+  isSelected: boolean;
+  generateButtonStyle: (isSelected: boolean) => React.CSSProperties;
+  isLinkSelected: (link: string) => boolean;
+}
+
+interface SubheaderListProps {
+  title: string;
+  buttons: ListButtonProps[];
+  generateButtonStyle: (isSelected: boolean) => React.CSSProperties;
+  isLinkSelected: (link: string) => boolean;
+}
+
+export function SubheaderList({
+  title,
+  buttons,
+  generateButtonStyle,
+  isLinkSelected,
+}: SubheaderListProps) {
+  const commonStyle = {
+    fontWeight: 'bold',
+    bgcolor: 'black',
+    color: 'white',
+    fontSize: 20,
+    textAlign: 'right',
+  };
```

Comment:
🔧 When we have a variable like this that has no dependency on the component props, it's better to take it out of the component definition